app/modules/user/components/signup.svelte

```diff
@@ -0,0 +1,142 @@
+  import { testPassword } from '#user/lib/password_tests'
+  import { verifyUsername } from '#user/lib/username_tests'
+
+  export let redirect = null
+
+  let username, email, password
+  let usernameFlash, emailFlash, signupFlash
+  let usernameInputNode, emailInputNode
+
+  async function earlyVerifyUsername () {
+    try {
+      if (username) await verifyUsername(username)
+    } catch (err) {
+      usernameFlash = err
+    }
+  }
+
+  async function earlyVerifyEmail () {
+    try {
+      if (email) await verifyEmail(email)
+    } catch (err) {
+      emailFlash = err
+    }
+  }
+
+  let signingUp = false
+  async function signup (e) {
+    // Prevent the button click to trigger the <form> action
+    e.preventDefault()
+    try {
+      signingUp = true
+      // Taking values directly from input to avoid missing changes triggered
+      // by password managers/addons/user scripts rather than user input
+      username = usernameInputNode.value.trim()
+      email = emailInputNode.value.trim()
+      await verifyUsername(username)
+      await verifyEmail(email)
+      testPassword(password)
+      await requestSignup({ username, email, password })
+      location.href = redirect || '/'
+    } catch (err) {
+      signupFlash = err
+    } finally {
+      signingUp = false
+    }
+  }
+</script>
+
+<div class="auth-menu">
+  <!-- Use a <form> to make the button be clicked on Enter and trigger to please password managers -->
+  <form
+    class="custom-cell"
+    method="post"
+    autocomplete="on"
+    autocorrect="off"
+    autocapitalize="off"
+  >
+    <h2 class="subheader">{i18n('sign up')}</h2>
+
+    <label for="username">{i18n('username')}</label>
+    <div class="input-box">
+      <input
+        type="text"
+        name="username"
+        id="username"
+        required
+        bind:value={username}
+        bind:this={usernameInputNode}
+        on:blur={earlyVerifyUsername}
+        on:keydown={() => usernameFlash = null}
+        use:autofocus
+      />
+      <Flash state={usernameFlash} />
+    </div>
+
+    <label for="email">{i18n('email')}</label>
+    <div class="input-box">
+      <input
+        type="text"
+        name="email"
+        id="email"
+        required
+        bind:value={email}
+        bind:this={emailInputNode}
+        on:blur={earlyVerifyEmail}
+        on:keydown={() => emailFlash = null}
+      />
+      <Flash state={emailFlash} />
+    </div>
+
+    <PasswordInput
+      bind:password
+      title={I18n('password')}
+      autocomplete="on"
+    />
+
+    <Flash state={signupFlash} />
+
+    <button
+      id="signup"
+      class="button"
+      on:click={signup}
+      disabled={signingUp}
+    >
+      {i18n('sign up')}
+      {#if signingUp}<Spinner />{/if}
+    </button>
+
+    <hr />
+
+    <div class="other-options">
+      <!-- TODO: improve button contrast #a11y -->
+      <a id="login" href="/login" on:click={loadInternalLink}>{I18n('login')}</a>
```

Comment:
I understand this login button is legacy, but it looks weird to me to have this green (valid) button which links to a diffirent context without any help/description of what this button is doing here. I checked a few services used by inventaire (wikidata, weblate, github) and none have a login link on their signup page.

I would be for removing it or having a description label such as ("Already have an account ?") above the button, and/or make the button a simple link like "Forgot password" on login page.

Reply:
I gave that a try in b593f934d